packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -78,6 +78,69 @@ class MemoryBodyState extends State<MemoryBody> {
     );
   }
 
+  static const String _liveFeed = 'Live Feed';
+  String memorySource = _liveFeed;
+
+  Widget createMenuItem(String name) {
+    final rowChildren = memorySource == name
+        ? [
+            Icon(Icons.check, size: 12),
+            const SizedBox(width: 10),
+            Text(name),
+          ]
+        : [
+            const SizedBox(width: 22),
+            Text(name),
+          ];
+
+    return PopupMenuItem<String>(
+      value: name,
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: rowChildren,
+      ),
+    );
+  }
+
+  Widget _selectMemoryFile() {
+    final List<String> files = _controller.memoryLog.offlineFiles();
+
+    final List<PopupMenuItem<String>> items = [
+      createMenuItem(_liveFeed),
+    ];
+
+    for (var index = 0; index < files.length; index++) {
+      items.add(createMenuItem(files[index]));
+    }
+
+    return PopupMenuButton<String>(
+      onSelected: (value) {
+        setState(() {
+          memorySource = value;
+
+          if (memorySource == _liveFeed) {
+            if (_controller.offline) {
+              // User is switching back to 'Live Feed'.
+              _controller.memoryTimeline.offflineData.clear();
+              _controller.offline = false; // We're live again...
+            } else {
+              // Still a live feed - keep collecting.
+              assert(!_controller.offline);
+            }
+          } else {
+            // Switching to an offline memory log (JSON file in /tmp).
+            _controller.memoryLog.loadOffline(memorySource);
+          }
+
+          // Notify the Chart state there's new data from a different memory
+          // source to plot.
+          _controller.notifyMemorySourceListeners();
```

Comment:
all of this code should be in the controller. `_controller.updateMemorySource(source)`

Reply:
All of the data and entry handling for MPChart is in the controller.